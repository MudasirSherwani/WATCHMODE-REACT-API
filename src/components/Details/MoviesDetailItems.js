import React from 'react';
import PropTypes from 'prop-types';
import './detailpage.css';

const MoviesTitleItems = (props) => {
  const {
    title,
    plotoverview,
    runtimeminutes,
    type,
    userrating,
    poster,
    trailer,
    releasedate,
    imdbid,
    usrating,
    year,
    originallanguage,
  } = props;

  return (
    <div className="table-container" data-testid="detailpagecontainer">
      <div className="detail-items">
        <h2 className="title">{title}</h2>
        <img className="poster" src={poster} alt="Movie Poster" />
        <a href={trailer} className="trailer">Watch Movie</a>
        <table className="table">
          <tbody>
            <tr>
              <th scope="col">Content Type</th>
              <td>{type}</td>
            </tr>
            <tr>
              <th scope="col">User Rating</th>
              <td>{userrating}</td>
            </tr>
            <tr>
              <th scope="col">Release Date</th>
              <td>{releasedate}</td>
            </tr>
            <tr>
              <th scope="col">Run Time</th>
              <td>
                {runtimeminutes}
                .0 min
              </td>
            </tr>
            <tr>
              <th scope="col">Imb ID</th>
              <td>{imdbid}</td>
            </tr>
            <tr>
              <th scope="col">Original Language</th>
              <td>{originallanguage}</td>
            </tr>
            <tr>
              <th scope="col">US Rating</th>
              <td>{usrating}</td>
            </tr>
            <tr>
              <th scope="col">Year</th>
              <td>{year}</td>
            </tr>
          </tbody>
        </table>
        <h3 className="desc">Descriptions</h3>
        <h4 className="desc">{plotoverview}</h4>
      </div>
    </div>
  );
};

MoviesTitleItems.propTypes = {
  title: PropTypes.string.isRequired,
};
export default MoviesTitleItems;
