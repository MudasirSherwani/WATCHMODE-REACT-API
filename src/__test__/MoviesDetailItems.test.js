import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom';
import store from '../redux/configureStore';
import MoviesDetailItems from '../components/Details/MoviesDetailItems';

it('Check Changes in the Detail Page', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Router>
        <Routes>
          <Route path="/" element={<MoviesDetailItems />} />
        </Routes>
      </Router>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Check rendering of div', () => {
  render(
    <Provider store={store}>
      <Router>
        <Routes>
          <Route path="/" element={<MoviesDetailItems title="title" />} />
        </Routes>
      </Router>
    </Provider>,

  );

  const detailPage = screen.getByTestId('detailpagecontainer');

  expect(detailPage).toBeInTheDocument();
});
