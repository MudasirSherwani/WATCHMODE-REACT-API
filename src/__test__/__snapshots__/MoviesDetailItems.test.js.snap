// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Changes in the Detail Page 1`] = `
<div
  className="table-container"
  data-testid="detailpagecontainer"
>
  <div
    className="detail-items"
  >
    <div
      className="header-data"
    >
      <img
        alt="Movie Poster"
        className="poster"
      />
      <div
        className="title-trailer"
      >
        <h2
          className="title"
        >
          Movie Name
        </h2>
        <a
          href="/"
          onClick={[Function]}
        >
          <button
            className="trailer"
            type="button"
          >
             Watch Movie! 
          </button>
        </a>
      </div>
    </div>
    <table
      className="table"
    >
      <tbody>
        <tr>
          <th
            scope="col"
          >
            Content Type
          </th>
          <td />
        </tr>
        <tr>
          <th
            scope="col"
          >
            User Rating
          </th>
          <td />
        </tr>
        <tr>
          <th
            scope="col"
          >
            Release Date
          </th>
          <td />
        </tr>
        <tr>
          <th
            scope="col"
          >
            Run Time
          </th>
          <td>
            .0 min
          </td>
        </tr>
        <tr>
          <th
            scope="col"
          >
            Imb ID
          </th>
          <td />
        </tr>
        <tr>
          <th
            scope="col"
          >
            Original Language
          </th>
          <td />
        </tr>
        <tr>
          <th
            scope="col"
          >
            US Rating
          </th>
          <td />
        </tr>
        <tr>
          <th
            scope="col"
          >
            Year
          </th>
          <td />
        </tr>
      </tbody>
    </table>
    <h3
      className="desc"
    >
      Descriptions
    </h3>
    <h4 />
  </div>
</div>
`;
